site/content/contribute/webapp/build-component.md

```diff
@@ -114,60 +117,50 @@ export default connect(mapStateToProps, mapDispatchToProps)(ItemList);
 
 If the selectors and/or actions you need don't yet exist in Redux then you should go add those first by following the [guide to adding actions and selectors](/contribute/redux/actions/).
 
-Your `index.js` and `item_list.js` files will live together in an `item_list/` directory.
+Your `index.ts` and `item_list.ts` files will live together in an `item_list/` directory.
 
-## Implementing your Component
+## Implementing the Component
 
-With the props defined and, if necessary, the container built you're ready to implement your component. For the most part, implementing a component for the webapp is no different than building any other React component. That said, we do have a few rules to abide by:
-
-1. All components must extend `React.PureComponent`.
-2. All data for rendering must come from props.
-3. Local rendering information that does not affect the store may be kept in state.
+With the props defined and, if necessary, the container built, you're ready to implement the rest of your component. For the most part, implementing a component for the web app is no different than building any other React component. While older code tends to use class components which inherit from `React.PureComponent`, most newer code should use functional components.
```

Comment:
```suggestion
With the props defined and, if necessary, the container built, you're ready to implement the rest of your component. For the most part, implementing a component for the web app is no different than building any other React component. While older code tends to use class components which extend from `React.PureComponent`, most newer code should use functional components.
```

Reply:
Just a slight tweak here since I don't think the "from" is needed if we switch to "extend"
```suggestion
With the props defined and, if necessary, the container built, you're ready to implement the rest of your component. For the most part, implementing a component for the web app is no different than building any other React component. While older code tends to use class components which extend `React.PureComponent`, most newer code should use functional components.
```